### DISCUSSION TITLE
Costificare Gas tramite uptime termostato

### DISCUSSION THREAD

**Discussion by DannyGi82** [indomus/forum#1367] [GENERALE - Discussioni generiche]
Ciao a tutti, ho acquistato un termostato wifi della garza smart che si appoggia a tuya.
Ho importato il tutto in Home assistant OS (virtualizzato tramite proxmox) tramite l'integrazione tuya e funziona perfettamente.
Avendo già un sensore per il consumo elettrico bticino che mi da le stats alla perfezione, vorrei integrare anche quello del termostato in modo da sapere quanto consumo di gas avviene.
La via che avevo pensato era di creare un sensore che usasse l'uptime del termostato, poi moltiplicarlo per il consumo orario della mia caldaia così da avere una stima dei costi.

```sensor:
  - platform: history_stats
    name: Uptime Riscaldamento
    entity_id: climate.riscaldamento
    state: "heat_cool"
    type: time
    start: "{{ now().replace(hour=0, minute=0, second=0) }}"
    end: "{{ now() }}"
```
Ora viene correttamente creato un sensore che mi da il tempo totale di accensione del termostato.
Come posso fare per trasformare questo in un sensore che mi dia m3?
Cioè io so che ogni ora la mia caldaia consuma 2,89 m3. Come posso creare un entità sensore che mi prenda il valore di sensor.climate.riscaldamento e lo moltiplichi per 2.89 con unità m3?

Grazie mille

**Comment by Marcoletto** [comment]
Ciao,
non credo possa funzionare come ragionamento.
Cioè, la caldaia non va sempre al 100%, almeno la mia.
Solitamente modula; magari all' inizio parte forte ma poi cala di potenza.
Quindi il consumo varia.

Inoltre dovresti mettere in conto l' utilizzo della caldaia per scaldare l' acqua sanitaria.

E' una mia idea eh, hai fatto queste valutazioni?
